### PR TITLE
Fixed https://github.com/wso2-extensions/esb-connector-salesforce/iss…

### PR DESCRIPTION
**Pull request by kesavany**
…ues/20

Add documentation.

## Purpose
Fix https://github.com/wso2-extensions/esb-connector-salesforce/issues/20
Add documentation.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: connector/docs/config.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+**Properties** 
+* username:  The user name to access the Salesforce account.
+* password:  The password provided here is a concatenation of the user password and the security token provided by Salesforce.
+* loginUrl:  The login URL to access the Salesforce account.
+* blocking:  Indicated whether the connector needs to perform blocking invocations to Salesforce. (Supported in ESB 4.9.0 and later) 
+
+```text
+*  Users can obtain the security token by changing the password or resetting the security token using the Salesforce user interface. The new security token will be sent to the email address recorded in the user's Salesforce record.
+*  The response of this operation is attached to the message body and is used for subsequent Salesforce operations.
+*  The session ID is saved into the property salesforce.sessionId, and the server URL is saved into salesforce.serviceUrl. If the given login details are invalid, the specified fault sequence will be triggered.
```
Comment (Nirdesha) [inline connector/docs/config.md]: The session ID is saved in the property salesforce.sessionId and the server URL is saved in salesforce.serviceUrl. If the given login details are invalid, the specified fault sequence is triggered.

File: connector/docs/config.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+The Salesforce connector operation examples use this convention to show how to specify the connection configuration for that operation. In all cases, the configKey attribute is optional if the connection to Salesforce has already been established and is required only if you need to specify a different connection from the current connection.
+
+## Common parameters
+
+Listed below are parameters that are common when [working with records](records.md) and [working with the recyclebin](recyclebin.md) in Salesforce.
+
+
+| Name | Description | Default value |
+| ------------- | ------------- | ------------- |
+| allowFieldTruncate | Set to 1, to truncate string values if they exceed the defined field length. | 0 |
```
Comment (Nirdesha) [inline connector/docs/config.md]: No comma:
Set to 1 to truncate string values

File: connector/docs/config.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+| allOrNone | Set to 1, to roll back changes if any object fails when multiple objects are sent.If set to 0 (false), some records can be processed successfully while others are marked as failed in the call results. | 0 |
```
Comment (Nirdesha) [inline connector/docs/config.md]: Set to 1 to roll back changes if any object fails when multiple objects are sent. If set to 0 (false), some records can be processed successfully while others are marked as failed in the call results.

Comment (kesavany) [reply connector/docs/config.md]: All comments fixed in https://github.com/wso2-extensions/esb-connector-salesforce/pull/23/files

**Pull request merged**